2.5/test/run

```diff
@@ -144,30 +146,55 @@ test_scl_usage() {
     echo "ERROR[exec /bin/sh -ic "${run_cmd}"] Expected '${expected}', got '${out}'"
     return 1
   fi
+  return 0
+}
+
+function test_scl_usage_functionality() {
+  test_scl_usage "ruby --version" "ruby ${RUBY_VERSION}."
+  check_result "$?" || return 1
+  return 0
+}
+
+function test_base_scl_usage() {
+  test_scl_usage
+  check_result "$?" || return 1
+  return 0
+}
+
+function test_application() {
+  # Verify that the HTTP connection can be established to test application container
+  run_test_application &
+
+  # Wait for the container to write it's CID file
+  wait_for_cid
+
+  test_connection
+  check_result "$?" || return 1
+  return 0
+}
+
+function test_npm_functionality() {
+  info "Testing npm availibility"
+  ct_npm_works
+  check_result "$?" || return 1
+  return 0
 }
 
 function run_all_tests() {
   local APP_NAME="$1"
   for test_case in $TEST_LIST; do
     info "Running test $test_case ...."
+    msg="for '${APP_NAME}' $test_case"
     if $test_case; then
-      printf -v test_short_summary "${test_short_summary}[PASSED] for '${APP_NAME}' $test_case\n"
+      printf -v test_short_summary "${test_short_summary}[PASSED] ${msg}\n"
     else
-      printf -v test_short_summary "${test_short_summary}[FAILED] for '${APP_NAME}' $test_case\n"
-      [ -n "${FAIL_QUICKLY:-}" ] && echo "$sum" && return 1
+      printf -v test_short_summary "${test_short_summary}[FAILED] ${msg}\n"
```

Comment:
My previous comments stand:

 - `code duplication` - can't the printf line can be put out the logic structure? (`if then else`)
    - only the PASSED / FAILED needs to be a variable, right?
 - note: you're not doing `cleanup` if you FAIL_QUICKLY; is that on purpose?
 - also I'm not sure why you're using `printf` in bash instead of `echo`
   - is it because the `test_short_summary`? Where does that get set? (Sorry I'm a bit lost here.)

Reply:
See comments here: https://github.com/sclorg/postgresql-container/pull/355/files

The same approach

Reply:
Let's leave it open. I will talk about it with team mates.

Reply:
printf is used because of `test_short_summary`.
You are right with FAILED_QUICKLY